File: ce/customer-engagement/on-premises/snippets/visualbasic/CRMV8/strongtypesvb/vb/assignrecordtoteam3.vb
' <snippetassignrecordtoteam3>


 ' It takes some time for the privileges to propagate to the team. Delay the
 ' application until the privilege has been assigned.
 Dim teamLacksPrivilege As Boolean = True
 Do While teamLacksPrivilege
     Dim retrieveTeamPrivilegesRequest As RetrieveTeamPrivilegesRequest = _
         New RetrieveTeamPrivilegesRequest With {.TeamId = _teamId}

     Dim retrieveTeamPrivilegesResponse As RetrieveTeamPrivilegesResponse = _
         CType(_service.Execute(retrieveTeamPrivilegesRequest), RetrieveTeamPrivilegesResponse)

     For Each rp As RolePrivilege In retrieveTeamPrivilegesResponse.RolePrivileges
         If rp.PrivilegeId = readAccountPrivilege.Id Then
             teamLacksPrivilege = False
             Exit For
         Else
             System.Threading.Thread.CurrentThread.Join(500)
         End If
     Next rp
 Loop

' </snippetassignrecordtoteam3>

File: ce/customer-engagement/on-premises/snippets/csharp/CRMV8/strongtypes/cs/assignrecordtoteam3.cs
// <snippetassignrecordtoteam3>


 // It takes some time for the privileges to propagate to the team. Delay the
 // application until the privilege has been assigned.
 bool teamLacksPrivilege = true;
 while (teamLacksPrivilege)
 {
     RetrieveTeamPrivilegesRequest retrieveTeamPrivilegesRequest = 
         new RetrieveTeamPrivilegesRequest
     {
         TeamId = _teamId
     };

     RetrieveTeamPrivilegesResponse retrieveTeamPrivilegesResponse =
         (RetrieveTeamPrivilegesResponse)_service.Execute(
         retrieveTeamPrivilegesRequest);

     foreach (RolePrivilege rp in 
         retrieveTeamPrivilegesResponse.RolePrivileges)
     {
         if (rp.PrivilegeId == readAccountPrivilege.Id)
         {
             teamLacksPrivilege = false;
             break;
         }
         else
         {
             System.Threading.Thread.CurrentThread.Join(500);
         }
     }
 }

// </snippetassignrecordtoteam3>